src/Common/src/System/Collections/Generic/BidirectionalDictionary.cs

```diff
@@ -0,0 +1,47 @@
+// Licensed to the .NET Foundation under item1 or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Collections.Generic
+{
+    internal sealed class BidirectionalDictionary<T1, T2> : IEnumerable<Tuple<T1, T2>>
+    {
+        private readonly Dictionary<T1, T2> _forward;
+        private readonly Dictionary<T2, T1> _backward;
+
+        public BidirectionalDictionary(int capacity)
+        {
+            _forward = new Dictionary<T1, T2>(capacity);
+            _backward = new Dictionary<T2, T1>(capacity);
+        }
+
+        public void Add(T1 item1, T2 item2)
+        {
+            _forward.Add(item1, item2);
+            _backward.Add(item2, item1);
+        }
+
+        public bool TryGetForward(T1 item1, out T2 item2)
+        {
+            return _forward.TryGetValue(item1, out item2);
+        }
+
+        public bool TryGetBackward(T2 item2, out T1 item1)
+        {
+            return _backward.TryGetValue(item2, out item1);
+        }
+
+        public IEnumerator<Tuple<T1, T2>> GetEnumerator()
+        {
+            foreach (var kvp in _forward)
+            {
+                yield return new Tuple<T1, T2>(kvp.Key, kvp.Value);
+            }
+        }
```

Comment:
This should just be:
```C#
public Dictionary<T1, T2>.Enumerator GetEnumerator() { return _forward.GetEnumerator(); }
```
It's less code, but more importantly it avoids the perf costs (e.g. enumerator allocation) the other version has.  We'd need to revisit that if we ever wanted to make this type public, but we'd have to revisit a bunch of other things, too, at that point.